cmd/kubeadm/app/constants/constants.go

```diff
@@ -446,36 +441,18 @@ func AddSelfHostedPrefix(componentName string) string {
 	return fmt.Sprintf("%s%s", SelfHostingPrefix, componentName)
 }
 
-// CreateTempDirForKubeadm is a function that creates a temporary directory under /etc/kubernetes/tmp (not using /tmp as that would potentially be dangerous)
-func CreateTempDirForKubeadm(dirName string) (string, error) {
-	tempDir := path.Join(KubernetesDir, TempDirForKubeadm)
-	// creates target folder if not already exists
-	if err := os.MkdirAll(tempDir, 0700); err != nil {
-		return "", errors.Wrapf(err, "failed to create directory %q", tempDir)
-	}
-
-	tempDir, err := ioutil.TempDir(tempDir, dirName)
+// CreateTempDirForKubeadm is a function that creates a temporary directory with name dirName in path prefix
+func CreateTempDirForKubeadm(prefix, dirName string) (string, error) {
+	tempDir, err := ioutil.TempDir(prefix, dirName)
```

Comment:
Just a suggestion, we should probably default prefix to KubernetesDir here. In that way you won't need to pass constants.KubernetesDir in too many places, plus you won't get an easy way to accidentally change the behavior (like the missing tmp part).

Reply:
The problem is how to default using a `string`, if it's `""` what should be passed to `TempDir`, `""` (what will trigger `os.TempDir()` prefix), or `KubernetesDir`? We would have the same issue again, right?

Reply:
```
if prefix == "" {
        prefix = path.Join(KubernetesDir, TempDirForKubeadm)
        // mkdir et all
}
```

The basic idea being, that the deleted block of code above gets into an if statement and s/tempDir/prefix (inside that if statement).

Test cases should then use TempDir to create the prefix dir for the test case.